From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 15 Mar 2011 05:35:24 +0000
Subject: [PATCH] ext4: Disable FS_IOC_FIEMAP ioctl temporarily

ext4 does not implement the fiemap operation correctly for extents
that are subject to delayed allocation.  A fix is pending, but until
that has been well-tested let's disable it.
---
 fs/ext4/file.c  |    2 ++
 fs/ext4/namei.c |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/fs/ext4/file.c b/fs/ext4/file.c
index 7b80d54..6b1b1ec 100644
--- a/fs/ext4/file.c
+++ b/fs/ext4/file.c
@@ -282,6 +282,8 @@ const struct inode_operations ext4_file_inode_operations = {
 	.removexattr	= generic_removexattr,
 #endif
 	.check_acl	= ext4_check_acl,
+#if 0
 	.fiemap		= ext4_fiemap,
+#endif
 };
 
diff --git a/fs/ext4/namei.c b/fs/ext4/namei.c
index 5485390..dc5961d 100644
--- a/fs/ext4/namei.c
+++ b/fs/ext4/namei.c
@@ -2539,7 +2539,9 @@ const struct inode_operations ext4_dir_inode_operations = {
 	.removexattr	= generic_removexattr,
 #endif
 	.check_acl	= ext4_check_acl,
+#if 0
 	.fiemap         = ext4_fiemap,
+#endif
 };
 
 const struct inode_operations ext4_special_inode_operations = {
-- 
1.7.4.1

